File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target)
	{
		Map<Integer, Integer> set = new HashMap<>();
		
		for (int i = 0; i < nums.length; i++)
		{
			int val = target - nums[i];
			if (set.containsKey(val))
				return new int[] {set.get(val), i};
			set.put(nums[i], i);
		}
		return new int[] {};
	}
}

File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}
        for i, val in enumerate(nums):
            if not dic.get(val):
                dic[target - val] = [val, i]
            else:
                return [dic.get(val)[1], i]
        return []